master/internal/configpolicy/task_config_policy.go

```diff
@@ -0,0 +1,51 @@
+package configpolicies
+
+import (
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
+)
+
+// ResourceConstraints are non-overridable resource constraints.
+// Submitted workloads that request resource quanities exceeding defined resource constraints in a
+// given scope are rejected.
+type ResourceConstraints struct {
```

Comment:
Resource constraints needs to look like this:
```
type ResourceConstraints struct{
    MaxSlots *int `json:"max_slots"`
}
```

Resource constraints is its own section in the user input. 
Something like
```
constraints:
  resources:
    max_slots: 4
  priority_limit: 20
  ```